source/server/options_impl_platform_linux.cc

```diff
@@ -0,0 +1,49 @@
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <bitset>
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <thread>
+
+#include "server/options_impl_platform.h"
+
+namespace Envoy {
+
+uint32_t OptionsImplPlatform::getCpuCount() {
+  unsigned int hw_threads = std::thread::hardware_concurrency();
+  std::string path("/proc/" + std::to_string(getpid()) + "/status");
+  std::string key("Cpus_allowed:");
+  std::ifstream file(path);
+  if (!file) {
+    // Fall back to number of hardware threads.
+    return hw_threads;
+  }
+
+  std::string statusLine;
```

Comment:
@oschaaf what are your thoughts on this vs. https://github.com/envoyproxy/envoy-perf/blob/master/nighthawk/source/common/utility.cc#L17?